snowballDevs/kandoo
# Implemented the feature to allow users to create new columns, delete columns

## Description

Enhancements and Refactoring for Column Management on the board

### New Column Creation:
- Implemented the ability for users to create new columns within the application.
- Sent column creation data to the backend for persistence, ensuring newly created columns now reside in the database.
- Enhanced the createColumn controller function to respond with the newly created column

### Deletion of a Column:
-  Implemented the ability for users to delete existing columns within the application.
- The selected column to delete is deleted in our db for persistence, ensuring columns are up to date in the db.

### Fixing of Routes 
- Column Routes in server was imported as tasks, fixed this.

## Related Issue

-   [ ] Is this related to a specific issue? Is so please specify:

## Type of Changes

Use a check for ✓ the following type of changes:

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓    | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before


### After
Successful Deletion of a column:
![image](https://github.com/snowballDevs/kandoo/assets/105207353/bc6e7b88-3992-4958-a5a5-491c49e729c5)

Successful creation of a column:
![image](https://github.com/snowballDevs/kandoo/assets/105207353/407f3cc0-b72a-41a2-8cd3-cb774e768e93)


## Testing Steps / QA Criteria

Now able to add, delete columns with the ui.
To check we are changing our state correctly please use React dev tools and look at the board context state after each action.


## Checklist:
